File: 283_Move_Zeros.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        slow, n = 0, len(nums)
        for fast in range(n):
            if nums[fast] == 0:
                continue
            if nums[fast] != nums[slow]:
                nums[slow] = nums[fast]
            slow += 1
                
        while slow < n:
            if nums[slow] != 0:
                nums[slow] = 0
            slow += 1
        return nums
    
# Two pointer approach made easier TC: O(N) SC: O(1)

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        slow = 0
        for fast in range(len(nums)):
            if nums[fast] !=0:
                nums[slow], nums[fast] = nums[fast], nums[slow]
                slow += 1

File: 283_Move_Zeros.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
    int numZeroes = 0;
    for (int i = 0; i < nums.size(); i++) {
        numZeroes += (nums[i] == 0);
    } 
        int i=0, j=0;
     while (i<nums.size() && j<nums.size()) {
        if (nums[j]!=0)
        {swap (nums[i], nums[j]); i++;}
        j++;
    }
   while (numZeroes) {
        nums[nums.size()-numZeroes]=0;
       numZeroes--;
        }  
    }
};